File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-jamba-development/JMBA_Kiosk_iOS/JambaKiosk/Reward.swift
//
//  Reward.swift
//  JambaJuice
//
//  Created by Eneko Alonso on 6/19/15.
//  Copyright (c) 2015 Jamba Juice. All rights reserved.
//

import Foundation
import SpendGoSDK
import OloSDK

struct Reward {

    // Shared fields (Olo & SpendGo)
    var name: String
    var type: String
    var desc: String
    var quantity: Int
    var value: Double

    // Olo specific fields
    var rewardId: Int64?
    var reference: String?
    var applied: Bool
    var membershipId: Int64?

    init(spendGoReward: SpendGoReward) {
        name = spendGoReward.rewardTitle
        desc = spendGoReward.desc
        quantity = spendGoReward.quantity
        type = spendGoReward.type
        applied = false
        value = 0
    }

    init(oloReward: OloLoyaltyReward) {
        name = oloReward.label
        desc = ""
        quantity = 1
        value = oloReward.value
        type = oloReward.type
        rewardId = oloReward.rewardId
        reference = oloReward.reference
        applied = oloReward.applied
        membershipId = oloReward.membershipId
    }

}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.2/JMBA_JambaJuice_Android/app/src/main/java/com/olo/jambajuice/BusinessLogic/Models/Reward.java
package com.olo.jambajuice.BusinessLogic.Models;

import com.wearehathway.apps.olo.Models.OloLoyaltyReward;
import com.wearehathway.apps.spendgo.Models.SpendGoReward;

import java.io.Serializable;

/**
 * Created by Ihsanulhaq on 5/27/2015.
 */
public class Reward implements Serializable
{
    // Shared fields (Olo & SpendGo)
    private String type;
    private String rewardTitle;
    private String description;
    int quantity;

    // Olo specific fields
    private int rewardId;
    private String reference;
    private boolean applied;
    private int membershipId;

    public Reward(SpendGoReward spendGoReward)
    {
        type = spendGoReward.getType();
        rewardTitle = spendGoReward.getReward_title();
        description = spendGoReward.getDescription();
        try
        {
            quantity = Integer.parseInt(spendGoReward.getQuantity());
        } catch (Exception e)
        {
        }

        applied = false;
    }

    public Reward(OloLoyaltyReward oloLoyaltyReward)
    {
        rewardId = oloLoyaltyReward.getRewardid();
        rewardTitle = oloLoyaltyReward.getLabel();
        description = "";
        type = oloLoyaltyReward.getType();
        applied = oloLoyaltyReward.isApplied();
        membershipId = oloLoyaltyReward.getMembershipid();
        reference = oloLoyaltyReward.getReference();
    }

    public String getDescription()
    {
        return description;
    }

    public String getRewardTitle()
    {
        return rewardTitle;
    }

    public String getType()
    {
        return type;
    }

    public int getRewardId()
    {
        return rewardId;
    }

    public String getReference()
    {
        return reference;
    }

    public boolean isApplied()
    {
        return applied;
    }

    public int getMembershipId()
    {
        return membershipId;
    }
}
